File: app/src/main/java/com/example/homeworkproject/adapter/ProvinceAdapter.java
package com.example.homeworkproject.adapter;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.homeworkproject.databinding.ProvinceLayoutBinding;
import com.example.homeworkproject.model.Province;

import java.util.ArrayList;


public class ProvinceAdapter extends RecyclerView.Adapter<ProvinceAdapter.ViewHolder> {


    private ArrayList<Province> provinceDataArrayList;


    public ProvinceAdapter() {

    }

    public void addList(ArrayList<Province> provinceList){
        this.provinceDataArrayList = provinceList;
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ProvinceAdapter.ViewHolder(ProvinceLayoutBinding.inflate(LayoutInflater.from(parent.getContext()),
                parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, @SuppressLint("RecyclerView") int position) {
        holder.itemView.province.setText(provinceDataArrayList.get(position).getProvinceName());

    }

    @Override
    public int getItemCount() {
        if (provinceDataArrayList == null){
            return 0;
        }
        return provinceDataArrayList.size();
    }


    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final ProvinceLayoutBinding itemView;

        public ViewHolder(@NonNull ProvinceLayoutBinding itemView) {
            super(itemView.getRoot());
            this.itemView = itemView;
        }

    }

}

File: app/src/main/java/com/example/homeworkproject/adapter/ProvinceAdapter.kt
package com.example.homeworkproject.adapter

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.homeworkproject.databinding.ProvinceLayoutBinding
import com.example.homeworkproject.model.Province

class ProvinceAdapter : RecyclerView.Adapter<ProvinceAdapter.ViewHolder>() {
    private var provinceDataArrayList: ArrayList<Province>? = null
    fun addList(provinceList: ArrayList<Province>?) {
        provinceDataArrayList = provinceList
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            ProvinceLayoutBinding.inflate(
                LayoutInflater.from(parent.context),
                parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, @SuppressLint("RecyclerView") position: Int) {
        holder.binding.province.text= provinceDataArrayList!![position].provinceName
    }

    override fun getItemCount(): Int {
        return if (provinceDataArrayList == null) {
            0
        } else provinceDataArrayList!!.size
    }

    inner class ViewHolder(val binding: ProvinceLayoutBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind (item : ProvinceLayoutBinding){
            with(binding) {
                 province.text
            }
        }
    }
}